cpp/src/datetime/datetime_ops.cu

```diff
@@ -169,6 +170,9 @@ std::unique_ptr<column> apply_datetime_op(column_view const& column,
       null_mask, column.null_count(),
       std::vector<std::unique_ptr<cudf::column>>{});
 
+  // Return an empty column if source column is empty
+  if (size == 0) return output;
```

Comment:
```suggestion
  if (column.size() == 0) return make_empty_column(output_col_type);
```

And move this line up right after the expects.

Reply:
done